src/core/prometheus.cc

```diff
@@ -217,9 +217,9 @@ static std::ostream& operator<<(std::ostream& os, const seastar::metrics::impl::
     switch (v.type()) {
     case seastar::metrics::impl::data_type::GAUGE:
     case seastar::metrics::impl::data_type::REAL_COUNTER:
-        return os << v.d();
+        return os << std::to_string(v.d());
```

Comment:
Looks like the standard is going to change to_string in c++26.

As discussed offline, perhaps:
```suggestion
        fmt::print(os, "{:.6f}", v.d());
        return os;
```
which may have the advantage of not allocating for large numbers.